yt_dlp/extractor/youtube.py

```diff
@@ -3443,7 +3442,9 @@ def extract_thread(contents, entity_payloads):
                         continue
                     comment_keys = list(filter(None, [view_model.get('commentKey'), view_model.get('toolbarStateKey')]))
                     entities = traverse_obj(entity_payloads, lambda _, v: v['entityKey'] in comment_keys)
-                    comment = self._extract_comment(view_model, entities, parent)
+                    comment = self._extract_comment(entities, parent)
+                    if comment:
+                        comment['is_pinned'] = traverse_obj(view_model, ('pinnedText', {str})) is not None
```

Comment:
About this.... Does this mean that there will be a key `is_pinned` for each and every single comment? I don't think this is necessary since only one comment will ever be pinned. Why have this key in every single comment?

Reply:
You are correct. The alternative to omit a non-truthy `is_pinned` field would be to do this:

```py
                    comment = self._extract_comment(entities, parent)
                    if comment and traverse_obj(view_model, ('pinnedText', {str})) is not None:
                        comment['is_pinned'] = True
```

Though the lack of a field or a `None` value typically should be reserved for when we don't have that data, rather than when we know it's `False`

Reply:
We should decide what fields to omit if-and-only-if the value is false, on the basis that the value is very rarely true. @pukkandan what's your opinion? I gave my rationale [earlier](https://github.com/yt-dlp/yt-dlp/pull/9775#discussion_r1602260893).

Reply:
> the lack of a field or a `None` value typically should be reserved for when we don't have that data, rather than when we know it's `False`

this


> We should decide what fields to omit if-and-only-if the value is false, on the basis that the value is very rarely true.

It cannot be done for any field since we want to be able to distinguish between not just True/False, but also a "Data not extracted" state

> It will bloat the final comments data.

Having "too much" data available has never been a concern we want to address. The infodict can be processed by third party scripts if that is an issue.

Reply:
Are there any outstanding issues left to address? The recent changes work for me.

Reply:
> Having "too much" data available has never been a concern we want to address. The infodict can be processed by third party scripts if that is an issue.

Agreed, while it is potentially "bloat", it's not really that much, specially since we talking about videos which are so much larger in size. Data-not-extracted state is helpful info to have, for further/future processing (health checks or something like that), if it would be ever a thing or someone makes a tool for it.